src/builder.rs

```diff
@@ -0,0 +1,141 @@
+    pub fn new() -> Self {
+        Self { files: vec![], schemas: vec![] }
+    }
+
+    /// Adds a configuration source with schema and optional parent directory
+    pub fn with_config_source<T: ToString, P: AsRef<Path>>(
+        mut self,
+        source: Source,
+        schema: T,
+        parent_dir: Option<P>,
+    ) -> Self {
+        let parent_dir = parent_dir.map(|p| p.as_ref().into());
+        self.schemas
+            .push(SchemaHolder { source, schema: schema.to_string(), parent_dir });
+        self
+    }
+
+    /// Adds an array of paths to configuration files
+    /// The file IO is carried out using runtime passed in build function
+    pub fn with_config_files<T: ToString>(mut self, files: &[T]) -> Self {
+        self.files
+            .push(files.iter().map(|v| v.to_string()).collect());
+        self
+    }
+
+    /// Builds TailcallExecutor with the provided runtime
+    pub async fn build(self, runtime: TargetRuntime) -> Result<TailcallExecutor> {
+        // Configuration reader initialization
+        let reader = ConfigReader::init(runtime.clone());
+
+        // Read configuration from files
+        let mut config_module = reader.read_all(&self.files).await?;
+
+        // Iterate over schema holders and merge configs
+        for holder in self.schemas {
+            let config = Config::from_source(holder.source, &holder.schema)?;
+            let new_config_module = reader.resolve(config, holder.parent_dir.as_deref()).await?;
+            config_module = config_module.merge_right(&new_config_module);
+        }
+
+        // Init AppContext
+        let blueprint = Blueprint::try_from(&config_module)?;
+        let app_ctx = AppContext::new(blueprint, runtime);
+        let app_ctx = Arc::new(app_ctx);
+        Ok(TailcallExecutor { config_module, app_ctx })
+    }
+}
+
+impl TailcallExecutor {
+    /// Validates configuration and performs optional schema validation
+    pub async fn validate(
+        &self,
+        n_plus_one_queries: bool,
+        schema: bool,
+        ops: Vec<OperationQuery>, // TODO: check if we can carryout IO for OperationQuery in this function
+    ) -> Result<String> {
+        log::info!("{}", "Config successfully validated".to_string());
+
+        let mut result_str = display_config(&self.config_module, n_plus_one_queries);
+        if schema {
+            let tbp = display_schema(&self.app_ctx.blueprint);
+            result_str = format!("{result_str}\n{tbp}");
+        }
+
+        validate_operations(&self.app_ctx.blueprint, ops)
+            .await
+            .to_result()?;
+
+        Ok(result_str)
+    }
+
+    /// Executes GraphQL request
+    pub async fn execute(self, req: Request<Body>) -> Result<Response<Body>> {
+        if self.app_ctx.blueprint.server.enable_batch_requests {
+            handle_request::<GraphQLBatchRequest>(req, self.app_ctx.clone()).await
+        } else {
+            handle_request::<GraphQLRequest>(req, self.app_ctx.clone()).await
```

Comment:
create a new builder module and move handle request logic inside builder. Reasoning — Builder should be created first and then the http server should use the builder to serve tailcall over http, here it looks a little opposite - Builder calling an http server API. Though that's not what actually happening, the module structure seems to communicate that it is.